crates/bin/sierra-compile-json/src/main.rs

```diff
@@ -28,22 +30,35 @@ struct Args {
 fn main() -> anyhow::Result<()> {
     let args = Args::parse();
 
-    let sierra_code = fs::read_to_string(args.file).with_context(|| "Could not read file!")?;
-    let Ok(sierra_program) = ProgramParser::new().parse(&sierra_code) else {
-        anyhow::bail!("Failed to parse Sierra program.")
-    };
-
-    let sierra_to_casm_config =
-        SierraToCasmConfig { gas_usage_check: args.gas_usage_check, max_bytecode_size: usize::MAX };
-
-    let cairo_program = compile(
-        &sierra_program,
-        &calc_metadata(&sierra_program, Default::default())
-            .with_context(|| "Failed calculating Sierra variables.")?,
-        sierra_to_casm_config,
-    )
-    .with_context(|| "Compilation failed.")?;
+    let file = std::fs::read(&args.file)?;
+    let filename = PathBuf::from(&args.file);
+    let sierra_program = match serde_json::from_slice(&file) {
+        Ok(program) => program,
+        Err(_) => {
+            let compiler_config = CompilerConfig {
+                replace_ids: true,
+                ..CompilerConfig::default()
+            };
+            let mut db = RootDatabase::builder()
+                .detect_corelib()
+                .skip_auto_withdraw_gas()
+                .build()
+                .unwrap();
+            let main_crate_ids = setup_project(&mut db, &filename).unwrap();
+            let sierra_program_with_dbg =
+                compile_prepared_db(&mut db, main_crate_ids, compiler_config).unwrap();
 
+            sierra_program_with_dbg.program
```

Comment:
This step is done by the command `cairo-compile`, here it should read the file as a textual Sierra file (*.sierra), being the output of the `cairo-compile --replace-ids` command.